pkg/pxc/app/binlogcollector/binlog_collector_test.go

```diff
@@ -0,0 +1,170 @@
+package binlogcollector
+
+import (
+	"reflect"
+	"testing"
+
+	api "github.com/percona/percona-xtradb-cluster-operator/pkg/apis/pxc/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestGetDeploymentLabelsAndMatchLabels(t *testing.T) {
+	createCR := func() *api.PerconaXtraDBCluster {
+		return &api.PerconaXtraDBCluster{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "test-cluster",
+				Namespace: "default",
+			},
+			Spec: api.PerconaXtraDBClusterSpec{
+				CRVersion: "1.15.0",
+				InitContainer: api.InitContainerSpec{
+					Resources: &corev1.ResourceRequirements{
+						Requests: nil,
+					},
+				},
+				PXC: &api.PXCSpec{
+					PodSpec: &api.PodSpec{
+						Enabled:                  true,
+						Size:                     3,
+						ContainerSecurityContext: &corev1.SecurityContext{},
+					},
+				},
+				Backup: &api.PXCScheduledBackup{
+					PITR: api.PITRSpec{
+						Enabled:            true,
+						StorageName:        "test-storage",
+						TimeBetweenUploads: 60.0,
+						TimeoutSeconds:     30.0,
+					},
+					Image:           "percona/pxc-backup:1.0",
+					ImagePullPolicy: corev1.PullAlways,
+					Storages: map[string]*api.BackupStorageSpec{
+						"test-storage": {
+							Type: "s3",
+							S3: &api.BackupStorageS3Spec{
+								Bucket:            "test-bucket",
+								Region:            "us-east-1",
+								CredentialsSecret: "minio-secret",
+								EndpointURL:       "https://minio-service.default:9000/",
+							},
+							Labels: map[string]string{
+								"storage-type": "s3",
+							},
+							Annotations: map[string]string{
+								"annotation-key": "annotation-value",
+							},
+							ContainerSecurityContext: &corev1.SecurityContext{},
+							PodSecurityContext:       &corev1.PodSecurityContext{},
+						},
+					},
+				},
+			},
+		}
+	}
+
+	tests := []struct {
+		name                string
+		cr                  *api.PerconaXtraDBCluster
+		existingMatchLabels map[string]string
+		expectedLabels      map[string]string
+		expectedMatchLabels map[string]string
+		expectError         bool
+	}{
+		{
+			name:                "Default labels and matchLabels",
```

Comment:
I think we need to assert the default case using the latest CR version, not the oldest. In the latest version, the expectedMatchLabels should be empty if the `existingMatchLabels ` is nil, is that correct?